controllers/lightspeed/openstacklightspeed_controller.go

```diff
@@ -0,0 +1,272 @@
+			condition.DeploymentReadyErrorMessage,
+			err.Error()))
+		return result, err
+	} else if (result != ctrl.Result{}) {
+		return result, nil
+	}
+
+	// NOTE: We cannot consume the OLSConfig definition directly from the OLS operator's code due to
+	// a conflict in Go versions. When this comment was written, the min. required Go version for
+	// openstack-operator was 1.21 whereas OLS operator required at least Go version 1.23. Once the
+	// Go versions catch up with each other we should consider consuming OLSConfig directly from OLS
+	// operator and updating this code and any subsequent code that consumes this structure.
+	olsConfig := uns.Unstructured{}
+	olsConfigGVK := schema.GroupVersionKind{
+		Group:   "ols.openshift.io",
+		Version: "v1alpha1",
+		Kind:    "OLSConfig",
+	}
+
+	olsConfig.SetGroupVersionKind(olsConfigGVK)
+	olsConfig.SetName(lightspeed.OLSConfigName)
+
+	_, err = controllerutil.CreateOrPatch(ctx, r.Client, &olsConfig, func() error {
+		// Check if the OpenStackLightspeed instance that is being processed owns the OLSConfig. If
+		// it is owned by other OpenStackLightspeed instance stop the reconciliation.
+		olsConfigLabels := olsConfig.GetLabels()
+		ownerLabel := ""
+		if val, ok := olsConfigLabels[lightspeed.OpenStackLightspeedOwnerIDLabel]; ok {
+			ownerLabel = val
+		}
+
+		if ownerLabel != "" && ownerLabel != string(instance.GetObjectMeta().GetUID()) {
+			return fmt.Errorf("OLSConfig is managed by different OpenStackLightspeed instance")
+		}
+
+		err = lightspeed.PatchOLSConfig(helper, instance, &olsConfig, indexID)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			lightspeedv1.OpenStackLightspeedReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			condition.DeploymentReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+
+	OLSConfigReady, err := lightspeed.IsOLSConfigReady(ctx, helper)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if OLSConfigReady {
+		instance.Status.Conditions.MarkTrue(
+			lightspeedv1.OpenStackLightspeedReadyCondition,
+			lightspeedv1.OpenStackLightspeedReadyMessage,
+		)
+	} else {
+		Log.Info("OLSConfig is not ready yet. Waiting.")
+		return ctrl.Result{RequeueAfter: time.Second * time.Duration(5)}, nil
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// reconcileDelete reconciles the deletion of OpenStackLightspeed instance
+func (r *OpenStackLightspeedReconciler) reconcileDelete(
+	ctx context.Context,
+	helper *common_helper.Helper,
+	instance *lightspeedv1beta1.OpenStackLightspeed,
+) (ctrl.Result, error) {
+	if ok := controllerutil.RemoveFinalizer(instance, helper.GetFinalizer()); !ok {
+		return ctrl.Result{}, fmt.Errorf("remove finalizer failed")
+	}
+
+	olsConfig, err := lightspeed.GetOLSConfig(ctx, helper)
+	if err != nil && !k8s_errors.IsNotFound(err) {
+		return ctrl.Result{}, err
+	} else if err != nil {
+		return ctrl.Result{}, nil
+	}
```

Comment:
not sure I understand this return. the above returns if there is an error and it is not `IsNotFound` . Do we have to return on `IsNotFound`?

I think what you want to do is to only run the bellow if no `IsNotFound` error happened. Otherwise you won't have an instance of `olsConfig`.

Reply:
Honestly, I do not understand it either in the hindsight. I updated the logic. Thank you!